### ISSUE TITLE
Issue with lets-encrypt script / DNS resolution

### ISSUE THREAD

**Issue by soleroitsolutionsltd** [nextcloud/vm#854]
## Steps to reproduce
1. Run /var/scripts/activate-ssl.sh

## Expected behaviour
Would get a enrolled in lets-encrypt 

## Actual behaviour
Error in script (see below)

## Server configuration

**Server OS** (Ubuntu server is default)
18.04.2

**How did you install the VM?** (Scripted install from master **OR** Released version)
Deployed OVA to ESX server

## Network
**Do you use DHCP?**
No

**Is port 80 and/or 443 open?**
Yes

## Logs / Screenshots
As you can see the script will try to resolve our FQDN using external resolver (vs the OS one) and fails...
---
Last login: Tue Jun 18 06:41:27 2019 from 172.16.110.27
 _  _         _      _             _
| \| |_____ _| |_ __| |___ _  _ __| |
| .` / -_) \ /  _/ _| / _ \ || / _` |
|_|\_\___/_\_\\__\__|_\___/\_,_\__,_|

https://www.hanssonit.se/nextcloud-vm

Hostname: nextcloud
WAN IPv4: 85.195.197.58
WAN IPv6:
LAN IPv4: 172.16.50.210

ncadmin@nextcloud:~$ sudo -i
[sudo] password for ncadmin:
root@nextcloud:~# cd /etc
root@nextcloud:/etc# cd scrips
-bash: cd: scrips: No such file or directory
root@nextcloud:/etc# cd scripts
-bash: cd: scripts: No such file or directory
root@nextcloud:/etc# /var/scripts/activate-ssl.sh
Are you sure you want to continue? ([y]es or [N]o): y
Have you forwarded port 80+443 in your router? ([y]es or [N]o): y
Do you have a domain that you will use? ([y]es or [N]o): y

+---------------------------------------------------------------+
|    Please enter the domain name you will use for Nextcloud:   |
|    Like this: example.com, or nextcloud.example.com           |
+---------------------------------------------------------------+

mycould.ffgg.com

Is this correct? mycould.ffgg.com ([y]es or [N]o): y

Checking if mycould.ffgg.com exists and is reachable...
Doing a DNS lookup for mycould.ffgg.com...
DNS lookup failed with nslookup.
Please check your DNS settings! Maybe the domain isn't propagated?
Please check https://www.whatsmydns.net/#A/mycould.ffgg.com if the IP seems correct.
Server:         9.9.9.9
Address:        9.9.9.9#53

** server can't find mycould.ffgg.com: NXDOMAIN

root@nextcloud:/etc# host mycloud.ffgg.com
mycloud.ffgg.com has address 172.16.50.210
root@nextcloud:/etc# dig mycloud.ffgg.com

; <<>> DiG 9.11.3-1ubuntu1.7-Ubuntu <<>> mycloud.ffgg.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 15471
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;mycloud.ffgg.com.              IN      A

;; ANSWER SECTION:
mycloud.ffgg.com.       3589    IN      A       172.16.50.210

;; Query time: 0 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Tue Jun 18 06:50:25 CEST 2019
;; MSG SIZE  rcvd: 61

**Comment by enoch85** [comment]
So, you are trying to run the SSL cript against a DNS record that are internal? That won't work. 

Please read this guide: https://docs.hanssonit.se/s/W6fMouPiqQz3_Mog/virtual-machines-vm/d/W6-83ePiqQz3_MrT/publish-your-server-online

It can take a while for a domain to propagate correctly, so please read the guide, make the necessary preparations, then try again.

**Comment by soleroitsolutionsltd** [comment]
> So, you are trying to run the SSL script against a DNS record that resolves to an internal IP? That won't work.
Nope
We are running what I believe is a fairly typically setup. Our server is on a NATed network with unroutable IPs (172.16.x.x).
As such we have internal DNS that will resolve FQDN to the internal Ip and external DNS that will resolve FQDN to the public IP. Post 80 and 443 are "natted".
This should be relatively obvious from the log I have posted:

```
Hostname: nextcloud
WAN IPv4: 85.195.197.58
WAN IPv6:
LAN IPv4: 172.16.50.210
```
Now when I run the script it will try to resolve our FQDN using 9.9.9.9
```
DNS lookup failed with nslookup.
Please check your DNS settings! Maybe the domain isn't propagated?
Please check https://www.whatsmydns.net/#A/mycould.ffgg.com if the IP seems correct.
Server: 9.9.9.9
```
which is **not** the machine resolver (I guess this is hard coded in the script ?). Doing so it will get the public IP - which does not match the host ip and the script fail with an error...

If I do a host or dig lookup it will use the machine configured resolver and find the local IP...

**Comment by enoch85** [comment]
Well, when I check https://www.whatsmydns.net/#A/mycould.ffgg.com it sure seems like it's not possible to resolve that address from the outside. 

![image](https://user-images.githubusercontent.com/4511254/59675485-a25b7b80-91c5-11e9-9dc2-8339a8b7d0f4.png)

The script uses 9.9.9.9 (which is hardcoded, yes) when resolving just for that matter - to check if the domain is publicly available. If not Let's Encrypt won't work as it resolves your *public* DNS. 

If you want to use SSL on your internal network or domain then you have to buy a certificate. Afaik Let's Encrypt doesn't support generating certificates for domains in your LAN.

I also use my own BIND config and NS servers for some domains, and the script works just fine.

**Comment by soleroitsolutionsltd** [comment]
Thanks
The host mame we want is mycloud.ffgg.com and it certainly resolves - there was a typo ! :(
```
MBP13:~ alex$ host mycloud.ffgg.com 9.9.9.9
Using domain server:
Name: 9.9.9.9
Address: 9.9.9.9#53
Aliases: 

mycloud.ffgg.com has address 85.195.197.57
MBP13
```

**Comment by enoch85** [comment]
OK, the issue is that the script won't let you continue cause the public IP of the host is: 85.195.197.58
While the DNS resolves 85.195.197.57. It must be the same.

> WAN IPv4: 85.195.197.58

https://github.com/nextcloud/vm/blob/master/lib.sh#L231
